articles/hdinsight/hbase/apache-hbase-advisor.md

```diff
@@ -0,0 +1,69 @@
+
+Ensure that the following configs are tuned
+1.  Set `hbase.rs.cacheblocksonwrite` to `true`  - This is by default true in HDInsight HBase. Ensure this not reset.
+2.  Increase the `hbase.hstore.compactionThreshold` value so that you can avoid the compaction from kicking in. By default this is `3`. You can increase it to a higher value like `10`.
+3. If you are doing #2 then change `hbase.hstore.compaction.max` to a higher value for eg `100` and also increase the value for the config `hbase.hstore.blockingStoreFiles` to higher value for eg `300`.
+4. If you are sure that you are interested only in the recent data then turn ON `hbase.rs.cachecompactedblocksonwrite` configuration so that even if compaction happens then we are sure that the data is in cache. These configs can be set at the family level also. In the hbase shell
+  ```
+   alter '<TableName>', {NAME => '<FamilyName>', CONFIGURATION => {'hbase.hstore.blockingStoreFiles' => '300'}}
+  ```
+5. Block cache can be turned off for a given family in a table. Ensure that it is turned ON for families that have most recent data reads. By default block cache is turned ON for all families in a table. In case you have disabled the block cache for a family and need to turn it ON then use the alter command from the hbase shell.
+
+
+The above configurations helps to ensure that as far as possible the data is in cache and that the recent data does not undergo compaction. If a TTL is possible in your usecase then mind considering Date tiered compaction. For more details on what is Date Tiered compaction check the below link,
+  ```
+   https://hbase.apache.org/book.html#ops.date.tiered
+  ```
+
+# Flush queue tuning
+The advisory may indicate that your flushes may need tuning. The flush handlers might not be enough.
+
+In the region server UI keep an eye on the flush queue and if it grows beyond 100 then it means that the flushes are slower and you may have to tune the  configuration  `hbase.hstore.flusher.count`. By default the value is 2. Ensure that the max flusher threads do not increase beyond 6.
+Also see if you have a recommendation for region count tuning. If so first try the region tuning to see if that helps in faster flushes.
+
+# Region count tuning
+The most prominent reason why you have blocked updates is that the region count might be more than the optimally supported heap size. Generally how to tune the heap size, memstore size and the region count? Let's see with an example
```

Comment:
Do we want to split the information on specific calculations under an example or more details section?